internal/resourcestore/resourcestore.go

```diff
@@ -181,20 +184,20 @@ func (rc *ResourceStore) Put(name string, resource IdentifiableCreatable, cleane
 // This is useful for situations where clients retry requests quickly after they "fail" because
 // they've taken too long. Adding a watcher allows the server to slow down the client, but still
 // return the resource in a timely manner once it's actually created.
-func (rc *ResourceStore) WatcherForResource(name string) chan struct{} {
+func (rc *ResourceStore) WatcherForResource(name string) (watcher chan struct{}, stage string) {
 	rc.mutex.Lock()
 	defer rc.mutex.Unlock()
-	watcher := make(chan struct{}, 1)
+	watcher = make(chan struct{}, 1)
 	r, ok := rc.resources[name]
 	if !ok {
 		rc.resources[name] = &Resource{
 			watchers: []chan struct{}{watcher},
 			name:     name,
```

Comment:
Perhaps you need to set the stage to unknown here.

Reply:
...and in every other place where you create a new `Resource`.

Alternatively, don't initialize it, but every time you return the stage value, check it for being empty and return `StageUnknown`. Is this is what you're doing, then nothing has to be changed in the code.